typescript/src/weighted/weightedPool.ts

```diff
@@ -8,13 +14,28 @@ import {
 
 export class Weighted implements PoolBase {
     public normalizedWeights: bigint[];
+    // Swap limits: amounts swapped may not be larger than this percentage of the total balance.
+    public _MAX_IN_RATIO = 300000000000000000n;
+    public _MAX_OUT_RATIO = 300000000000000000n;
 
     constructor(poolState: {
         weights: bigint[];
     }) {
         this.normalizedWeights = poolState.weights;
     }
 
+    getMaxSwapAmount(swapParams: MaxSwapParams): bigint {
```

Comment:
I'm not really sure the best way to organize things, but it feels like we're breaking a pattern here.
If we decide to keep all math within WeightedMath, then we could have a `computeMaxSwapAmount` method that can reuse `_MAX_IN_RATIO` already set there.
What do you think?

Reply:
I forgot we already had the MAX values defined in maths! I've use them as an import instead.
Lets keep `PoolMath` as close to the contracts as possible. The maxSwap calculation is kind of a custom requirement for our SOR use case (more so for non-weighted pools).